docs/JavaCodeQuality.md

```diff
@@ -521,6 +521,58 @@ private static final int SIXTIES_START = 1960; // good
+When suppressing warnings, document your reasoning and report false positives to the rule maintainers to help improve rule accuracy.  
+**Rule name:** UsingSuppressWarnings  
+**Example:**
+```java
+@SuppressWarnings("pmd:UnconditionalCreatedLogArguments") // Reason for suppression: the rule doesn't recognize record getters (yet), and those are not expensive
+public void process(ConsumerRecord consRecord) {
+    ConsumerEvent event = consRecord.event(); // record getter
+    log.info("Received event '{}'", event); // pmd:UnconditionalCreatedLogArguments suppressed
+}
+```
+**Note:** This rule is just informational to track use of SuppressWarnings, to be able to double-check if suppression is correct. It often doesn't need a fix.
+The @SuppressWarning is the preferred way to suppress a rule violation, because you can specify the exact rule you want to suppress. Downside is that it does
+not work on line level. Two other options are:
+* `// NOPMD`: suppresses all PMD violations on the line, not just the one you want to suppress.
+* `// NOSONAR`: suppresses all Sonar violations on the line, not just the one you want to suppress.
+The `// NOPMD` is also generated when you use the Intellij PMD Plugin and right click and select `Suppress`.
+It will specify the rule you want to suppress explicitly, and also ask you to specify a reason for the suppression.
+```java
+@SuppressWarnings({"pmd:AvoidConcatInAppend", "pmd:UsingSuppressWarnings"}) //NOPMD - suppressed UsingSuppressWarnings - TODO explain reason for suppression
+class Foo implements KeyGenerator {
+    // ...
+}
+```
+In this example you see that you can use it to suppress the SuppressWarnings rule itself. This is not a good idea, because it is not clear why you are suppressing the rule.
+The Intellij PMD Plugin will give issues on information level about the usage of `// NOPMD` so they can be easily reviewed.
+The `// NOSONAR` is not checked by both PMD Plugin and Sonar, so be careful with its usage.
+
+### M03
+
+**Observation: @SuppressWarnings is used for high risk problems**  
+**Problem:** The suppressed rule detects high risk problems, suppressing the detected violations without full knowledge can lead to incidents like customer data mix-up, corrupt data, server crashes or very bad performance.  
+**Solution:** Only suppress violations (warnings) when you have complete understanding of the rule, and you are sure it does not apply.
+When suppressing warnings, document your reasoning and report false positives to the rule maintainers to help improve rule accuracy.  
+**Rule name:** UsingSuppressWarningsHighRisk  
+**Note:** This rule is to track use of SuppressWarnings for high risk rule violations, to be able to double-check if suppression is correct. It doesn't necessarily need a fix.   
+**Example:**  
+```java
+@SuppressWarnings({"pmd:AvoidUnguardedMutableFieldsInSharedObjects", "PMD.AvoidUserDataInSharedObjects"}) // suspicious, bad to suppress
```

Comment:
suspicious, bad to suppress -> suspicious, **probably** bad to suppress (also in rule description?)